### PR TITLE
oded/add-examSrv

### PR DESCRIPTION
**Pull request by odeds**

### CODE REVIEW

File: src/components/exerciseUtility/services/category.service.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        };
+
+        self.getParentCategory = function (categoryId) {
+            return self.getCategoryMap().then(function (categories) {
+                var parentId = categories[categoryId].parentId;
+                return categories[parentId];
+            });
+        };
+
+        self.getSubjectIdByCategory = function (category) {
```
Comment (igormas) [inline src/components/exerciseUtility/services/category.service.js]: Please change to getCategoryLevel1Parent

File: src/components/exerciseUtility/services/category.service.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            if (category.typeId === categoryEnum.SUBJECT.enum) {
+                return $q.when(category.id);
+            }
+            return self.getParentCategory(category.id).then(function (parentCategory) {
+                return self.getSubjectIdByCategory(parentCategory);
+            });
+        };
+
+
+        self.getTestScore = function (categoryId) {
```
Comment (igormas) [inline src/components/exerciseUtility/services/category.service.js]: Please change to getCategoryLevel2Parent

File: src/components/exerciseUtility/services/category.service.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            return self.getCategoryMap().then(function (categories) {
+                var category = categories[categoryId];
+                if (categoryEnum.TEST_SCORE.enum === category.typeId) {
+                    return category;
+                }
+                return self.getTestScore(category.parentId);
+            });
+        };
+
+        self.getAllGeneralCategories = (function () {
```
Comment (igormas) [inline src/components/exerciseUtility/services/category.service.js]: Please change to getAllLevel3Categories

File: src/components/exerciseUtility/services/category.service.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                            }
+                        });
+                        return generalCategories;
+                    });
+                }
+                return getAllGeneralCategoriesProm;
+            };
+        })();
+
+        self.getAllGeneralCategoriesBySubjectId = (function () {
```
Comment (igormas) [inline src/components/exerciseUtility/services/category.service.js]: Please change to getAllLevel3CategoriesGroupedByLevel1

File: src/components/exerciseUtility/services/category.service.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                        return $q.all(promArray).then(function () {
+                            return generalCategories;
+                        });
+                    });
+                }
+                return getAllGeneralCategoriesBySubjectIdProm;
+            };
+        })();
+
+        self.getAllSpecificCategories = (function () {
```
Comment (igormas) [inline src/components/exerciseUtility/services/category.service.js]: Please change to getAllLevel4Categories

File: src/components/scoring/scoring.service.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            var data = _getResultsFn(scoreTable, questionsResults, typeId, subjectId);
+            return {
+                sectionScore: data
+            };
+        }
+
+        // api
+
+        scoringServiceObjApi.isTypeFull = function (typeId) {
+            return ExamTypeEnum['FULL TEST'].enum === typeId;
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: Please change the 'FULL TEST' enum name to 'FULL_TEST'

File: dist/main.js [hunk 7]
```diff
@@ -3250,6 +3546,143 @@ angular.module('znk.infra.popUp').run(['$templateCache', function($templateCache
...
+            };
+        }
+
+        // api
+
+        scoringServiceObjApi.isTypeFull = function (typeId) {
+            return ExamTypeEnum['FULL TEST'].enum === typeId;
+        };
+
+        scoringServiceObjApi.getTestScoreResult = function (questionsResults, typeId, categoryId) {
```
Comment (igormas) [inline dist/main.js]: Its should not be here since it belong to SAT project

File: src/components/scoring/scoring.service.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+            if (angular.isDefined(id)) {
+                data = data[id];
+            }
+            if (angular.isDefined(rawScore)) {
+                data = data[rawScore];
+            }
+            return data;
+        }
+
+        function _getResultsFn(scoreTable, questionsResults, typeId, id) {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: change id parameter to subjectId

File: src/components/scoring/scoring.service.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+            var rawScore = _getRawScore(questionsResults);
+            var key = _getScoreTableKeyByTypeId(typeId);
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: please change to scoreTableKey

File: src/components/scoring/scoring.service.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+        function _isTypeFull(typeId) {
+            return ExamTypeEnum['FULL TEST'].enum === typeId;
+        }
+
+        function _getScoreTableKeyByTypeId(typeId) {
+            return _isTypeFull(typeId) ? keysMapConst.test : keysMapConst.miniTest;
+        }
+
+        function _getDataFromTable(scoreTable, key, id, rawScore) {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: please change key -> scoreTableKey , id -> subjectId

File: src/components/scoring/scoring.service.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+            };
+        }
+
+        // api
+
+        scoringServiceObjApi.isTypeFull = function (typeId) {
+            return ExamTypeEnum['FULL TEST'].enum === typeId;
+        };
+
+        scoringServiceObjApi.getTestScoreResult = function (questionsResults, typeId, categoryId) {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: is it exam score ?

File: src/components/scoring/scoring.service.js [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+            return _getScoreTableProm().then(function (scoreTable) {
+                return _getTestScoreResultFn(scoreTable, questionsResults, typeId, categoryId);
+            });
+        };
+
+        scoringServiceObjApi.getSectionScoreResult = function (questionsResults, typeId, subjectId) {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: Please change to getSectionScore

File: src/components/scoring/scoring.service.js [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+        };
+
+        scoringServiceObjApi.rawScoreToScore = function (subjectId, rawScore) {
+            return _getScoreTableProm().then(function (scoreTable) {
+                var roundedRawScore = Math.round(rawScore);
+                return _getDataFromTable(scoreTable, keysMapConst.test, subjectId, roundedRawScore);
+            });
+        };
+
+        scoringServiceObjApi.getTotalScoreResult = function (scoresArr) {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: Please change to getExamScore

File: src/components/scoring/scoring.service.js [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+            var totalScores = 0;
+            angular.forEach(scoresArr, function (score) {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: Wrong calculation, should be injected via config phase

File: src/components/scoring/scoring.service.js [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+                totalScores += score;
+            });
+            return $q.when(totalScores);
+        };
+
+        scoringServiceObjApi.getScoringSettings = function() {
```
Comment (igormas) [inline src/components/scoring/scoring.service.js]: Why the settings exposed ?

Comment (odeds) [inline src/components/scoring/scoring.service.js]: im not using the settings in the scoring service in any way, the only ones who use it, it's other components like diagnostic and userGoals for example

File: src/components/scoring/scoring.service.js (hunk 13 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: changed

File: src/components/scoring/scoring.service.js (hunk 12 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: changed

File: src/components/scoring/scoring.service.js (hunk 10 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: changed

File: src/components/scoring/scoring.service.js (hunk 9 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: changed

File: src/components/scoring/scoring.service.js (hunk 8 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: it gets category id or subject id so i think should stay id

File: src/components/scoring/scoring.service.js (hunk 6 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: changed

File: src/components/exerciseUtility/services/category.service.js (hunk 5 above)
Comment (odeds) [inline src/components/exerciseUtility/services/category.service.js]: changed

File: src/components/exerciseUtility/services/category.service.js (hunk 3 above)
Comment (odeds) [inline src/components/exerciseUtility/services/category.service.js]: changed

File: src/components/exerciseUtility/services/category.service.js (hunk 2 above)
Comment (odeds) [inline src/components/exerciseUtility/services/category.service.js]: changed

File: src/components/exerciseUtility/services/category.service.js (hunk 1 above)
Comment (odeds) [inline src/components/exerciseUtility/services/category.service.js]: changed

File: src/components/scoring/scoring.service.js (hunk 14 above)
Comment (odeds) [inline src/components/scoring/scoring.service.js]: done

File: src/components/exerciseUtility/services/category.service.js (hunk 4 above)
Comment (odeds) [inline src/components/exerciseUtility/services/category.service.js]: changed

File: src/components/scoring/scoring.service.js (hunk 8 above)
Comment (igormas) [inline src/components/scoring/scoring.service.js]: So change to categoryId

**Pull request merged**